src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -537,7 +550,7 @@ module.exports = class LevelPlane {
   getBlocksToPush(position, offsetX = 0, offsetY = 0) {
     let pushingBlocks = [];
     let workingPosition = position;
-    while (this.inBounds(workingPosition) && this.getBlockAt(workingPosition).blockType !== "") {
+    while (this.inBounds(workingPosition) && this.getBlockAt(workingPosition).blockType !== "" && !this.getBlockAt(workingPosition).blockType.startsWith("redstone") && !this.getBlockAt(workingPosition).blockType.startsWith("door")) {
```

Comment:
as a general rule, when logic about the identity of a block gets this complex it should really go in a method on LevelBlock itself, then we can just call `this.getBlockAt(workingPosition).getIsPushable()`

Reply:
👍